src/sdk/clients/createBicoPaymasterClient.test.ts

```diff
@@ -133,4 +141,144 @@ describe.runIf(paymasterTruthy())("bico.paymaster", async () => {
     // No gas fees were paid, so the balance should have decreased only by 1n
     expect(finalBalance).toBe(initialBalance - 1n)
   })
+
+  test("should use token paymaster to pay for gas fees, use max approval", async () => {
+    const nexusClient = await createNexusClient({
+      signer: account,
+      chain,
+      paymaster: createBicoPaymasterClient({
+        transport: http(paymasterUrl)
+      }),
+      paymasterContext: {
+        mode: "ERC20",
+        tokenInfo: {
+          feeTokenAddress: baseSepoliaUSDCAddress
+        }
+      },
+      transport: http(),
+      bundlerTransport: http(bundlerUrl),
+      ...testParams
+    })
+
+    const initialBalance = await publicClient.getBalance({
+      address: nexusAccountAddress
+    })
+
+    const hash = await nexusClient.sendTransaction({
+      to: recipientAddress,
+      value: 1n,
+      chain: baseSepolia
+    })
+
+    // Wait for the transaction to be mined
+    const { status } = await publicClient.waitForTransactionReceipt({ hash })
+    expect(status).toBe("success")
+    // Get final balance
+    const finalBalance = await publicClient.getBalance({
+      address: nexusAccountAddress
+    })
+
+    // Check that the balance hasn't changed
+    // No gas fees were paid, so the balance should have decreased only by 1n
+    expect(finalBalance).toBe(initialBalance - 1n)
+  })
+
+  test("should use token paymaster to pay for gas fees, use custom approval with token paymaster quotes", async () => {
+    const nexusClient = await createNexusClient({
+      signer: account,
+      chain,
+      index: 3n,
+      paymaster: createBicoPaymasterClient({
+        transport: http(paymasterUrl)
+      }),
+      paymasterContext: {
+        mode: "ERC20",
+        tokenInfo: {
+          feeTokenAddress: baseSepoliaUSDCAddress
+        }
+      },
+      transport: http(),
+      bundlerTransport: http(bundlerUrl),
+      ...testParams
+    })
+
+    const usdcBalance = await publicClient.readContract({
+      address: baseSepoliaUSDCAddress,
+      abi: parseAbi([
+        "function balanceOf(address owner) public view returns (uint256 balance)"
+      ]),
+      functionName: "balanceOf",
+      args: [nexusClient.account.address]
+    })
+    expect(usdcBalance).toBeGreaterThan(0n)
+
+    const initialBalance = await publicClient.getBalance({
+      address: nexusClient.account.address
+    })
+
+    const tokenList = [baseSepoliaUSDCAddress]
+    const userOp = await nexusClient.prepareUserOperation({
+      calls: [
+        {
+          to: recipientAddress,
+          value: 1n,
+          chain: baseSepolia
+        }
+      ]
+    })
+    const quote = await paymaster.getTokenPaymasterQuotes(userOp, tokenList)
+    const usdcFeeAmount = parseUnits(
+      quote.feeQuotes[0].maxGasFee.toString(),
+      quote.feeQuotes[0].decimal
+    )
+
+    expect(usdcBalance).toBeGreaterThan(usdcFeeAmount)
+
+    const hash = await nexusClient.sendTransaction(
```

Comment:
What if a user wants to use sendUserOperation() instead of sendTransaction()?

Reply:
I think writing a `prepareTokenPaymasterUserOperation()` call would be a better thing than attempting to hijack what sendTransaction does here. 

Then later you could add a `sendTokenPaymasterUserOperation()` which will pair `prepareTokenPaymasterUserOperation` with (sendTransaction() or `sendUserOperation`)

Reply:
I don't think this is a good idea. We are adding too many methods, this will become confusing and hard to use. What is the issue with using sendTransaction like this ? A user uses the same method he is used to. It doesn't make sense to add separated methods to send token paymaster user ops, we don't do this for sponsored mode.

For "sendUserOperation" the user (developer) will have to make sure he also batches an approval inside the user op if the smart account does not have enough allowance. Calling the "sendUserOperation" assumes you want to built the user op yourself, for some reason, so this means you also need to make sure the user op is valid. "sendTransaction" is an abstraction on top of user operations, we should encourage everyone to use this and only use "sendUserOperation" for special cases.

Coming back to your idea, I can add an "prepareTokenPaymasterUserOperation" or we can keep the familiar DEVX and use the already existing "prepareUserOperation" with an extra boolean parameter "withTokenPaymaster", if true our SDK injects an approve tx inside the user op, if needed.

Reply:
If you write a 'prepareTokenPaymasterUserOperation' helper then it should be available for use without changes via either sendTransaction pr sendUserOperation. In fact that would make for a good test to write:

("should use nexusClient.prepareTokenPaymasterUserOperation to prepare a user op, and then send it using nexusClient.sendUserOperation", () => {})

Reply:
sendTransaction doesn't do much more than simply call sendUserOperation. It's only really there as a way of keeping some form of backwards compatibility for our v4 users